src/reference/binaries.md

```diff
@@ -0,0 +1,118 @@
+   - [Running from cloned GitHub repository](#source-run).
+- [Using the pre-built Docker images](#docker-install).
+
+
+## Using the Source GitHub Repository {#source}
+
+To perform any of the actions using the GitHub repository (i.e., [building & installing](#source-install), or [running](#source-run) binaries), first install the [Rust toolchain](https://www.rust-lang.org/tools/install).
+
+### Installing from Source GitHub Repository {#source-install}
+
+You can install Iroha 2 binaries system-wide with the Rust `cargo` tool:
+
+::: code-group
+
+```shell [irohad]
+cargo install --git https://github.com/hyperledger-iroha/iroha.git irohad --locked
+irohad --help
+```
+```shell [iroha]
+cargo install --git https://github.com/hyperledger-iroha/iroha.git iroha_cli --locked
+iroha --help
+```
+
+```shell [kagami]
+cargo install --git https://github.com/hyperledger-iroha/iroha.git iroha_kagami --locked
+kagami --help
+```
+:::
+
+::: tip
+For more details on `cargo install` and its `[options]`, see [The Cargo Book > cargo-install(1)](https://doc.rust-lang.org/cargo/commands/cargo-install.html).
+:::
+
+### Running from Cloned GitHub Repository {#source-run}
+
+You may want to run the binaries without installing them system-wide. This will involve cloning Iroha's source code and using `cargo run`.
+
+First, clone the [hyperledger-iroha/iroha](https://github.com/hyperledger-iroha/iroha.git) GitHub repository:
+```shell
+git clone https://github.com/hyperledger-iroha/iroha.git <clone-folder>
+cd <clone-folder>
+```
+
+To run an Iroha 2 binary with `cargo install`, execute one of the following commands in your terminal:
+
+::: code-group
+
+```shell [irohad]
+cargo run --release --bin irohad -- --help
+```
+
+```shell [iroha]
+cargo run --release --bin iroha_cli -- --help
+```
+
+```shell [kagami]
+cargo run --release --bin iroha_kagami -- --help
+```
+
+:::
+
+::: tip
+
+For more details on `cargo run` and its `[options]`, see [The Cargo Book > cargo-run(1)](https://doc.rust-lang.org/cargo/commands/cargo-run.html).
+
+:::
+
+## Using the Pre-Built Docker Images {#docker-install}
+
+First, install the following Docker prerequisites:
+
+- [Docker](https://docs.docker.com/get-docker/)
+
+To install a binary from a pre-built Docker image, execute one of the following commands:
+
+::: code-group
+
+```shell [irohad]
+docker run -t hyperledger/iroha:dev
+```
+
+```shell [iroha]
+docker run -t hyperledger/iroha:dev iroha
+```
+
+```shell [kagami]
+docker run -t hyperledger/iroha:dev kagami
+```
+
+:::
+
+`irohad` and `iroha` binaries require config files that need to be made available to the container and access to host machine network connection.
+
+Below is an example of how to launch `iroha` with a your custom config and network access.
+
+```shell
+docker run --network host -v /path/to/host/config:/path/in/container/config -t hyperledger/iroha:dev iroha --config /path/in/container/config/client.toml
```

Comment:
```suggestion
docker run --network host -v /path/to/host/config:/config -t hyperledger/iroha:dev iroha --config /config/custom-client-config.toml
```

Config directory in the container is fixed in `Dockerfile` (https://github.com/hyperledger-iroha/iroha/blob/2c5e0555056e14c1bad5db6f700363ceea20843c/Dockerfile#L28)

Reply:
Updated in commit 894fe1e2024db529371d0d0ba9f1fecea2a56a06
Went into a bit more detail there.